test_positive_synchronize_docker_repo_set_tags_later :E AssertionError: [] is not None: E AssertionError: KeyError not raised

self = <tests.foreman.api.test_repository.DockerRepositoryTestCase testMethod=test_positive_synchronize_docker_repo_set_tags_later>

    @run_only_on('sat')
    @tier2
    def test_positive_synchronize_docker_repo_set_tags_later(self):
        """Verify that adding tags whitelist and re-syncing after
        synchronizing full repository doesn't remove content that was
        already pulled in
    
        :id: 6838e152-5fd9-4f25-ae04-67760571f6ba
    
        :expectedresults: Non-whitelisted tags are not removed
        """
        tags = ['latest']
        product = entities.Product(organization=self.org).create()
        repo = entities.Repository(
                content_type='docker',
                docker_upstream_name='alpine',
                name=gen_string('alphanumeric', 10),
                product=product,
                url=DOCKER_REGISTRY_HUB,
        ).create()
        repo.sync()
        repo = repo.read()
> self.assertIsNone(repo.docker_tags_whitelist)
E AssertionError: [] is not None

tests/foreman/api/test_repository.py:1505: AssertionError
2019-03-06 13:56:43


self = <tests.foreman.cli.test_repository.RepositoryTestCase testMethod=test_positive_synchronize_docker_repo_set_tags_later>

    @run_only_on('sat')
    @tier2
    def test_positive_synchronize_docker_repo_set_tags_later(self):
        """Verify that adding tags whitelist and re-syncing after
        synchronizing full repository doesn't remove content that was
        already pulled in
    
        :id: 97f2087f-6041-4242-8b7c-be53c68f46ff
    
        :expectedresults: Non-whitelisted tags are not removed
        """
        tags = 'latest'
        repo = self._make_repository({
            'content-type': 'docker',
            'docker-upstream-name': 'hello-world',
            'url': DOCKER_REGISTRY_HUB,
        })
        Repository.synchronize({'id': repo['id']})
        repo = self._validated_image_tags_count(repo=repo)
        # Field is displayed only if there is any content
        with self.assertRaises(KeyError):
> repo['container-image-tags-filter']
E AssertionError: KeyError not raised

tests/foreman/cli/test_repository.py:1054: AssertionError

--- a/tests/foreman/api/test_repository.py
+++ b/tests/foreman/api/test_repository.py
@@ -1548,7 +1548,7 @@ class DockerRepositoryTestCase(APITestCase):
         ).create()
         repo.sync()
         repo = repo.read()
-        self.assertIsNone(repo.docker_tags_whitelist)
+        self.assertEqual(len(repo.docker_tags_whitelist), 0)
         self.assertGreaterEqual(repo.content_counts['docker_tag'], 2)
 
         repo.docker_tags_whitelist = tags

--- a/tests/foreman/cli/test_repository.py
+++ b/tests/foreman/cli/test_repository.py
@@ -1049,9 +1049,7 @@ class RepositoryTestCase(CLITestCase):
         })
         Repository.synchronize({'id': repo['id']})
         repo = self._validated_image_tags_count(repo=repo)
-        # Field is displayed only if there is any content
-        with self.assertRaises(KeyError):
-            repo['container-image-tags-filter']
+        self.assertFalse(repo['container-image-tags-filter'])
         self.assertGreaterEqual(int(repo['content-counts']
                                     ['container-image-tags']), 2)
         Repository.update({

